builtin/joined_dataset.cc

```diff
@@ -299,10 +170,18 @@ struct JoinedDataset::Itl
                     // This has xxx results
                     //gotRow(...);
 
-                    RowName leftName(res->values.at(0).toUtf8String());
-                    RowName rightName(res->values.at(2).toUtf8String());
+                    //cerr << "got rows complex " << res->values.size() << endl;
+                    Utf8String leftNameUtf8 = res->values.at(0).toUtf8String();
+                    size_t i = 2;
+                    for (; i+ 2 < res->values.size(); i+=2)
+                    {
+                        leftNameUtf8 += "-" + res->values.at(i).toUtf8String();
+                    }                        
 
-                    gotRow(leftName, leftName, rightName, rightName);
+                    RowName leftName(leftNameUtf8);
+                    RowName rightName(res->values.at(i).toUtf8String());
+
+                    recordJoinRow(leftName, leftName, rightName, rightName);
```

Comment:
In many (other) cases we already have the rowHash so we'll avoid computing it twice